lambdaclass/mirra_backend
# [GH-798] - Configurator Game Modes

Closes #798 

## Motivation

Add the game mode concept to the configurator. So we can group configurations below a game mode


<img src="https://github.com/user-attachments/assets/e18e39a3-ddd4-4326-baba-985231ce930c" align="center">


DISCLAIMER: We're not addressing UI/UX concerns in this PR.
 
## Summary of changes

- [Add game modes table](https://github.com/lambdaclass/mirra_backend/pull/855/commits/2af4a799abbbe597ea1503743c7ace9264eb585e)
- [Add relationship between versions and game modes](https://github.com/lambdaclass/mirra_backend/pull/855/commits/b164d1b85d000fc12c97bdaa4cee885c46c0ff36)
- [Add mvc for game modes](https://github.com/lambdaclass/mirra_backend/pull/855/commits/178aa09282db72c010ab630ce37d00e88b5ee9e1)
- [Fix seeds. They were broken due to the game modes addition](https://github.com/lambdaclass/mirra_backend/pull/855/commits/34e1babd1ff040007f3390123fc96b52f37ef13a)
- [Add tests for the game modes schema](https://github.com/lambdaclass/mirra_backend/pull/855/commits/3aae697b35291c79a1ea2732cc3166d5e0f0341d)
- [Add game modes routes](https://github.com/lambdaclass/mirra_backend/pull/855/commits/6a13455d7233d0cf9cb71a571f014e8775920e43)
- [Add queries for game mode logic](https://github.com/lambdaclass/mirra_backend/pull/855/commits/1b7cb68805592585c9a82898a0d95a58ebde80c1)
- [Add endpoint and needed logic to retrieve the current config in a gamemode](https://github.com/lambdaclass/mirra_backend/pull/855/commits/d3ad3dc238ee563b4d86f1ed32f7a0da9f75c390)
- [Add placeholder for game mode name](https://github.com/lambdaclass/mirra_backend/pull/855/commits/c555f535a7c47c81833758436c488e1364d2f207)
- [Arena gets the configuration by game mode and not by version](https://github.com/lambdaclass/mirra_backend/pull/855/commits/144108845e3a63c3af65323a6c1eb4e1000f7e23)
- [Remove no longer used endpoints for configuration](https://github.com/lambdaclass/mirra_backend/pull/855/commits/4f51409e35c4d3da00f4324d45289174d0fe1a17)
- [Show game mode in version's show](https://github.com/lambdaclass/mirra_backend/pull/855/commits/cee752f2ae55062855180e00562f980a611c0cb7)
- [Refactor get_current_version_from_game_mode/1 function](https://github.com/lambdaclass/mirra_backend/pull/855/commits/77c5b01bb019bb0c09826da85aa5f9aed6164a3b)
- [Fix feature mark as current version when no former version exists](https://github.com/lambdaclass/mirra_backend/pull/855/commits/86450a9b6d166a438b03b1c7b8cf4fda3ab64322)
- [Add description field to game modes](https://github.com/lambdaclass/mirra_backend/pull/855/commits/f4b39ca608f328b10fad7b3320b6474705fdc91b)
## How to test it?

UI/UX stills rough.

- Go to `CONFIGURATOR_URL/game_modes` create a game mode
- Go to `CONFIGURATOR_URL/versions` and select a game mode
- Play a Game and check that everything works as usually

## Checklist
